File: 844-backspace-string-compare/JavaScript/2-backspace-string-compare[brute-force].js
const s = 'ab#c';
t = 'ad#c';

const buildString = (string) => {
  const buildString = [];
  for (let p = 0; p < string.length; p++) {
    string[p] !== '#' ? buildString.push(string[p]) : buildString.pop();
  }

  return buildString.join('').toString();
};

var backspaceCompare = function (s, t) {
  const finalS = buildString(s);
  const finalT = buildString(t);

  return finalS === finalT;
};

console.log(backspaceCompare(s, t));

File: 844-backspace-string-compare/Python/2-backspace-string-compare[brute-force].py
s = 'ab#c'
t = 'ad#c'

class Solution:
    def build_string(self,string):
      built_string = []
      for s in string:
        if s != '#':
          built_string.append(s)
        elif built_string:
          built_string.pop()
      return built_string.join('')

    def backspaceCompare(self, s: str, t: str) -> bool:
      final_s = self.build_string(s)
      final_t = self.build_string(t)

      return final_s == final_t



solution = Solution()

print(solution.backspaceCompare(s, t))